File: CSharpPrototype/Timer.cs
namespace Work_Timer_Box
{
    public class Timer
    {
        public bool IsRunning { get; protected set; }
        public long StartTime { get; protected set; }

        public virtual void StartTimer()
        {
            if (!IsRunning)
            {
                IsRunning = true;
                StartTime = Box.CurrentTime;
            }
        }

        public virtual long GetElapsed()
        {
            return StartTime == 0 ? 0 : Box.CurrentTime - StartTime;
        }

        public virtual long GetElapsedMinutes()
        {
            return UnixTimeToMinutes(GetElapsed());
        }

        public virtual long GetElapsedSeconds()
        {
            return UnixTimeToSeconds(GetElapsed());
        }

        public virtual void ResetTimer()
        {
            if (IsRunning)
            {
                IsRunning = false;
                StartTime = 0;
            }
        }


        public static long UnixTimeToMinutes(long time)
        {
            return time / 60;
        }

        public static long UnixTimeToSeconds(long time)
        {
            return time % 60;
        }
    }
}

File: Pico/timer.cpp
#include "consulting_clock.hpp"
#include "pico/time.h"

void Timer::StartTimer()
{
    if (!isRunning)
    {
        isRunning = true;
        startTime = time_us_64();
    }
}

uint64_t Timer::GetElapsed()
{
    return startTime == 0 ? 0 : time_us_64() - startTime;
}

uint16_t Timer::GetElapsedMinutes()
{
    return Timer::GetElapsed() / US_TO_MIN;
}

uint16_t Timer::GetElapsedSeconds()
{
    return Timer::GetElapsed() % US_TO_MIN / US_TO_SEC;
}

void Timer::ResetTimer()
{
    if (isRunning)
    {
        isRunning = false;
        startTime = 0;
    }
}

void Timer::RestartTimer()
{
    startTime = 0;
}
